plugins/generator-1.19.4/forge-1.19.4/templates/livingentity/livingentity.java.ftl

```diff
@@ -604,17 +604,21 @@ public class ${name}Entity extends ${extendsClass} <#if data.ranged>implements R
     </#if>
 
 	<#if data.waterMob>
-	@Override public boolean canBreatheUnderwater() {
-    	return true;
-    }
-
-    @Override public boolean checkSpawnObstruction(LevelReader world) {
+	@Override public boolean checkSpawnObstruction(LevelReader world) {
 		return world.isUnobstructed(this);
 	}
+	</#if>
 
-    @Override public boolean isPushedByFluid() {
-		return false;
-    }
+	<#if data.breatheUnderwater?? && (hasProcedure(data.breatheUnderwater) || data.breatheUnderwater.getFixedValue())>
+	@Override public boolean canBreatheUnderwater() {
+		return <@procedureOBJToConditionCode data.breatheUnderwater true false/>;

```

Comment:
Shouldn't this procedure have default value of `false` and the other one of `true`?

Reply:
If no procedure is defined, upper if checks for true, for "data.breatheUnderwater.getFixedValue()" and if that is true, default value should be true too.

This looks correct to me